src/fairing.rs

```diff
@@ -11,6 +16,10 @@ lazy_static! {
         MediaType::parse_flexible("application/wasm").unwrap(),
         MediaType::parse_flexible("application/octet-stream").unwrap(),
     ];
+    static ref CACHED_FILES: Mutex<HashMap<(String, bool, bool), (Vec<u8>, String)>> = {
```

Comment:
Probably doesn't matter a ton for perf, but I think it would be good to use a `RwLock` here, and preferably an async-friendly one.  Since we're pulling in tokio anyway, can use [`tokio::sync::RwLock`](https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html) here.

I've seen some comments from people saying that using blocking synchronization primitives in async code isn't really bad, but I've always been suspicious of it personally.  If you feel strongly though lmk

Reply:
I did find some performance issues in this implementation related to read/write locking. [I did already find a solution using `drop` on the Mutex values, which I could also port to this)](https://github.com/JonLatane/jonline/commit/315b2c57866b442bce41d207e72a14cdcfa79264#diff-6dee0eac1d219581ddc0daeeb48e4f8b81d7e8eebcdf73a448ca884be4ded5db). It looks like `RwLock` basically does this explicitly which is cool :) (I'm learning Rust and its frameworks with this pet project, if that's not clear!) I'll look into updating to use the appropriate `tokio` package and this primitive shortly though! (Hopefully will have an update tomorrow as I have evening plans.)